### PR TITLE
lint: run goheader enforcing standardized Go header template

### PR DESCRIPTION
**Pull request by ti-mo**
Since we didn't have a linter to enforce a particular Go file header format, this patch adds one with a relatively simple structure.

goheader can only match on a single block of // statements, so anything that cannot be encoded in the template needed to be moved to a separate block.

For Hubble packages, 'Authors of Hubble' was kept as the primary tagline. Any other copyright notices were moved to subsequent blocks.

Fixes: https://github.com/cilium/cilium/pull/20769#issuecomment-1281718208

```release-note
Fix up and lint SPDX headers in all Go files
```

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
:astonished:

**Review by rolinh — APPROVED** [review]

File: .golangci.yaml [hunk 1]
```diff
@@ -49,17 +49,19 @@ linters-settings:
       - nilness
   goimports:
     local-prefixes: github.com/cilium/cilium
-  misspell:
-    # Correct spellings using locale preferences for US or UK.
-    # Default is to use a neutral variety of English.
-    # Setting locale to US will correct the British spelling of 'colour' to 'color'.
-    # locale: US
-    ignore-words:
-      - rela # This is for elf.SHT_RELA
```
Comment (joestringer) [inline .golangci.yaml:58]: Was this change intentional?

File: pkg/bpf/binary/binary.go [hunk 2]
```diff
@@ -1,33 +1,7 @@
+// SPDX-License-Identifier: BSD-3-Clause
 // Copyright Authors of Cilium
-// SPDX-License-Identifier: Apache-2.0
-//
-// Copyright (c) 2009 The Go Authors. All rights reserved.
-//
-// Redistribution and use in source and binary forms, with or without
-// modification, are permitted provided that the following conditions are
-// met:
```
Comment (joestringer) [inline pkg/bpf/binary/binary.go]: I think the intent behind the previous arrangement for this code was that we had relicensed this code Apache-2.0, but it is still subject to the original license. Given the wording here, I don't think we can remove this licensing text.

**Review by joestringer — CHANGES_REQUESTED** [review]

File: .golangci.yaml (hunk 1 above)
Comment (ti-mo) [reply .golangci.yaml:58]: It was. iirc the code this was necessary for was removed with the alignchecker rewrite. Does this need to go into a separate commit?

**Review by christarazi — APPROVED** [review]

File: pkg/bpf/binary/binary.go (hunk 2 above)
Comment (ti-mo) [reply pkg/bpf/binary/binary.go]: Ah, so it was relicensed after all; I'll change it back. Didn't know if Apache2 and BSD licences were compatible.

The licensing text has been removed in correspondence to the upstream file, see commit msg. The full headers are no longer included.

File: .golangci.yaml (hunk 1 above)
Comment (joestringer) [reply .golangci.yaml:58]: Ideally yeah, in particular I asked the question because the commit message didn't mention it at all, so I wasn't sure how it related to the linting changes.

File: pkg/bpf/binary/binary.go (hunk 2 above)
Comment (joestringer) [reply pkg/bpf/binary/binary.go]: Ah I see, in that case I think the changes here are fine. Could you also link the upstream location in the commit message so it's easier to verify this?

**Review by joestringer — APPROVED** [review]
Based on the discussions on my prior feedback, I don't think any changes to the content are necessary. It'd be nice to update the commit messages to highlight the background to the changes but otherwise this LGTM. Thanks for this, it's caught a bunch of missing licenses in source files :tada:

**Review by qmonnet — APPROVED** [review]

File: .golangci.yaml (hunk 1 above)
Comment (ti-mo) [reply .golangci.yaml:58]: Sure thing! :+1:

**Review by aspsk — APPROVED** [review]

File: pkg/bpf/binary/binary.go (hunk 2 above)
Comment (ti-mo) [reply pkg/bpf/binary/binary.go]: Sure thing! Commit messages are not documentation, so I've added a reference to the path of the original files in the comments.

**Comment by ti-mo** [comment]
Hit some unrelated unit test flake in Travis, marking ready.
